matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TimelineChunk.kt

```diff
@@ -133,9 +133,27 @@ internal class TimelineChunk(private val chunkEntity: ChunkEntity,
             val prevEvents = prevChunk?.builtItems(includesNext = false, includesPrev = true).orEmpty()
             deepBuiltItems.addAll(prevEvents)
         }
+
+        if (timelineSettings.isLiveSenderInfo) {
+            updateToLiveSenderData(deepBuiltItems)
+        }
+
         return deepBuiltItems
     }
 
+    private fun updateToLiveSenderData(deepBuiltItems: ArrayList<TimelineEvent>) {
```

Comment:
We do not use `ArrayList` type nowadays. Can you replace with `List`.
Also since `SenderInfo` has to be immutable, I think this method should return an updated list.

Reply:
Done. Thank you.